### PR TITLE
[New feature] Add the alias and show it as column name when querying

### PR DESCRIPTION
**Pull request by Alima777**
select s1 as d1_temperature form root.sg.d1

The result will be like:

| Time | d1_temperature |
|   ...    |            ....             | 

select count(s1) as s1_num from root.sg.d1

The result will be like:

|    s1_num    |
|         ....        | 

You can see the detail:

[Add the alias and show it as column name when querying](https://lists.apache.org/thread.html/r1bdafb36efe0e05bd0f105a81f29e46f2251def192a1af5070160ab7%40%3Cdev.iotdb.apache.org%3E)

### CODE REVIEW

**Review by Genius-pig — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java [hunk 1]
```diff
@@ -664,11 +674,9 @@ private void deduplicate(QueryPlan queryPlan) throws MetadataException {
     if (queryPlan instanceof LastQueryPlan) {
       for (int i = 0; i < paths.size(); i++) {
         Path path = paths.get(i);
-        String column;
-        if (path.getAlias() != null) {
-          column = path.getFullPathWithAlias();
-        } else {
-          column = path.toString();
+        String column = path.getTsAlias() != null ? path.getTsAlias() : null;
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java:677]:
```suggestion
        String column = path.getTsAlias();
```

File: server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java [hunk 2]
```diff
@@ -689,14 +697,14 @@ private void deduplicate(QueryPlan queryPlan) throws MetadataException {
...
-      String column;
-      if (indexedPath.left.getAlias() != null) {
-        column = indexedPath.left.getFullPathWithAlias();
-      } else {
-        column = indexedPath.left.toString();
-      }
-      if (queryPlan instanceof AggregationPlan) {
-        column = queryPlan.getAggregations().get(indexedPath.right) + "(" + column + ")";
+      // judge whether as clause is used or not first
+      String column = indexedPath.left.getTsAlias() != null ? indexedPath.left.getTsAlias() : null;
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java:701]:
```suggestion
      String column = indexedPath.left.getTsAlias();
```

File: server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java [hunk 3]
```diff
@@ -788,11 +787,15 @@ private void getWideQueryHeaders(
           }
         }
         for (int i = 0; i < paths.size(); i++) {
-          if (paths.get(i).getAlias() != null) {
-            respColumns.add(aggregations.get(i) + "(" + paths.get(i).getFullPathWithAlias() + ")");
-          } else {
-            respColumns.add(aggregations.get(i) + "(" + paths.get(i).getFullPath() + ")");
+          Path path = paths.get(i);
+          // judge whether as clause is used or not first
+          String column = path.getTsAlias() != null ? path.getTsAlias() : null;
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java:792]:
```suggestion
          String column = path.getTsAlias();
```

File: server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java (hunk 1 above)
Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java]: Sorry...Fixed.

File: server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java (hunk 2 above)
Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java]: Fixed.

File: server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java (hunk 3 above)
Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java]: Fixed.

**Review by JackieTien97 — APPROVED** [review]
Good job.

File: server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java [hunk 4]
```diff
@@ -761,22 +759,23 @@ private TSExecuteStatementResp getQueryColumnHeaders(PhysicalPlan physicalPlan,
...
     switch (plan.getOperatorType()) {
       case QUERY:
       case FILL:
         for (Path path : paths) {
-          if (path.getAlias() != null) {
-            respColumns.add(path.getFullPathWithAlias());
-          } else {
-            respColumns.add(path.getFullPath());
+          // judge whether as clause is used or not first
+          String column = path.getTsAlias() != null ? path.getTsAlias() : null;
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java:772]:
```suggestion
          // judge whether as clause is used or not first
          String column = path.getTsAlias();
```

Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java]: fixed

File: server/src/main/java/org/apache/iotdb/db/metadata/PartialPath.java [hunk 5]
```diff
@@ -31,7 +31,10 @@
 public class PartialPath extends Path implements Comparable<Path> {
 
   private String[] nodes;
-  private String alias;
+  // alias of sensor
+  private String alias = null;
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/metadata/PartialPath.java:35]: measurementAlias

File: server/src/main/java/org/apache/iotdb/db/metadata/PartialPath.java [hunk 6]
```diff
@@ -31,7 +31,10 @@
 public class PartialPath extends Path implements Comparable<Path> {
 
   private String[] nodes;
-  private String alias;
+  // alias of sensor
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/metadata/PartialPath.java:34]:
```suggestion
  // alias of measurement
```

File: server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java [hunk 7]
```diff
@@ -690,14 +700,14 @@ private void deduplicate(QueryPlan queryPlan) throws MetadataException {
...
     for (Pair<PartialPath, Integer> indexedPath : indexedPaths) {
-      String column;
-      if (indexedPath.left.getAlias() != null) {
-        column = indexedPath.left.getFullPathWithAlias();
-      } else {
-        column = indexedPath.left.getFullPath();
-      }
-      if (queryPlan instanceof AggregationPlan) {
-        column = queryPlan.getAggregations().get(indexedPath.right) + "(" + column + ")";
+      // judge whether as clause is used or not first
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java:703]: what is not first?

**Review by qiaojialin — APPROVED** [review]
Good work, please fix the ci and some minor naming

File: server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java (hunk 7 above)
Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java:703]: If as clause is used, measurementAlias will be covered.

File: server/src/main/java/org/apache/iotdb/db/metadata/PartialPath.java (hunk 5 above)
Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/metadata/PartialPath.java:35]: Fixed.

File: server/src/main/java/org/apache/iotdb/db/metadata/PartialPath.java (hunk 6 above)
Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/metadata/PartialPath.java:34]: Fixed.

**Pull request merged**